### DISCUSSION TITLE
Use unified code style for Python?

### DISCUSSION THREAD

**Discussion by cmeessen** [research-software-directory/RSD-as-a-service#187] [General]
For Python, there is no unified code style being used at the moment, at least as far as I can see. I think in the future the code might benefit from a style, as the overall looks will be the same, no matter who committed code.

My suggestion would be to use [Python Black](https://github.com/psf/black), which is rather strict and has the advantage that it comes with an auto formatter. We have had good experiences with Black so far.

When committing to a style, adding a step that checks for style violations in the CI/CD would make sense too.

What are your opinions about that?

**Comment by jmaassen** [comment]
Indeed having a unified code style is a very good idea. The python we use at the moment is inherited from the previous RSD version. There are some changes here and there to adapt it to the new service. I would be happy to transform this into the python black style.

**Reply by cmeessen** [reply]
Coming back to @ewan-escience's question about tabs, this was discussed quite emotionally on their github (see [here](https://github.com/psf/black/issues/47) or [here](https://github.com/psf/black/issues/2798)). Their argument is that PEP8 suggests usage of four-space indentation, and because this appears to be the de-facto standard. I don't mind using tabs, in that case it seems [yapf](https://github.com/google/yapf) is the preferred formatter.

**Reply by ewan-escience** [reply]
yapf seems ok to me. In general I think we need the following features when looking for a formatter and yapf seems to have them all:
- a command line interface that can output whether or not the code is formatted properly, so we can use it in CI/CD
- a command line interface than can format the code
- it is configurable with a config file

It is only for Python, though. We should look for tools for other languages (including Java, JavaScript, TypeScript, HTML, CSS) as well.
